mantatail.py

```diff
@@ -48,15 +60,24 @@ def delete_user(self, nick: str) -> None:
         Note: This does not actually disconnect the user from the server.
         To disconnect the user, a tuple (None, None) must be put in their send queue.
         """
-        user = self.connected_users[nick.lower()]
+        user = self.find_user(nick)
+        assert user is not None
+
         for channel in self.channels.values():
             if user in channel.users:
                 channel.users.discard(user)
         del self.connected_users[nick.lower()]
 
     def delete_channel(self, channel_name: str) -> None:
-        """Removes a channel from server."""
-        del self.channels[channel_name.lower()]
+        """
+        Removes a channel from server.
+        Silently ignores a non-existing channel.
+        """
+        try:
+            del self.channels[channel_name.lower()]
+        except KeyError as err:
+            print(err)
+            print("Tried to delete a channel that does not exist.")
```

Comment:
Just let Python handle printing an error message. It does better job than you did. The problem is that `print(err)` shows a very short error message, such as `'#mantatest'`, whereas Python by default would show multiple lines ending with `KeyError: '#mantatest'`.

For example:

```
>>> channels = {}
>>> try:
...     del channels["#mantatest"]
... except KeyError as err:
...     print(err)
... 
'#mantatest'
```

```suggestion
        del self.channels[channel_name.lower()]
```

Reply:
However, wouldn't the default Python error break/interrupt the server, whereas in an `except, the server keeps running? That's how I thought it worked.

Reply:
If you really want the server to continue running, even though its internal state seems to be broken in an unexpected way, you can use `traceback.print_exc()` to print a full error message. But that should be done in a separate PR.

Reply:
Okay, would you say it makes more sense to just let the server break if this error occurs?

Reply:
I would much rather have the server interrupt itself than e.g. let non-operators ban other users, or do something else that the server might end up doing if the state is messed up.